From 48e1a3a4a1e9336afc3868e42bc75c90131270a7 Mon Sep 17 00:00:00 2001
From: Jaegeuk Kim <jaegeuk@motorola.com>
Date: Tue, 2 Jun 2015 14:38:23 -0700
Subject: [PATCH 773/959] IKSWL-41251 - block: add real written data size in
 /sys/block/<dev>/stat

The trim size in sectors is added into sectors[WRITE] in /sys/block/<dev>/stat.
So, this patch introduces 12'th index showing actual written data size .

Change-Id: Ife9e3576a213b260ba3c5c956035e6c989efb194
Signed-off-by: Jaegeuk Kim <jaegeuk@motorola.com>
Reviewed-on: http://gerrit.mot.com/753797
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Christopher Fries <cfries@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
Reviewed-on: http://gerrit.mot.com/831630
Reviewed-by: Igor Kovalenko <igork@motorola.com>
---
 block/blk-core.c          | 2 ++
 block/partition-generic.c | 5 +++--
 include/linux/genhd.h     | 1 +
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/block/blk-core.c b/block/blk-core.c
index bce8d736b41..8cb300649f7 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -2137,6 +2137,8 @@ static void blk_account_io_completion(struct request *req, unsigned int bytes)
 		cpu = part_stat_lock();
 		part = req->part;
 		part_stat_add(cpu, part, sectors[rw], bytes >> 9);
+		if (rw == WRITE && !(req->cmd_flags & REQ_DISCARD))
+			part_stat_add(cpu, part, data_sectors, bytes >> 9);
 		part_stat_unlock();
 	}
 }
diff --git a/block/partition-generic.c b/block/partition-generic.c
index 47284e71265..387c8d03136 100644
--- a/block/partition-generic.c
+++ b/block/partition-generic.c
@@ -120,7 +120,7 @@ ssize_t part_stat_show(struct device *dev,
 	return sprintf(buf,
 		"%8lu %8lu %8llu %8u "
 		"%8lu %8lu %8llu %8u "
-		"%8u %8u %8u"
+		"%8u %8u %8u %8llu"
 		"\n",
 		part_stat_read(p, ios[READ]),
 		part_stat_read(p, merges[READ]),
@@ -132,7 +132,8 @@ ssize_t part_stat_show(struct device *dev,
 		jiffies_to_msecs(part_stat_read(p, ticks[WRITE])),
 		part_in_flight(p),
 		jiffies_to_msecs(part_stat_read(p, io_ticks)),
-		jiffies_to_msecs(part_stat_read(p, time_in_queue)));
+		jiffies_to_msecs(part_stat_read(p, time_in_queue)),
+		(unsigned long long)part_stat_read(p, data_sectors));
 }
 
 ssize_t part_inflight_show(struct device *dev,
diff --git a/include/linux/genhd.h b/include/linux/genhd.h
index ec274e0f4ed..02f323c97d2 100644
--- a/include/linux/genhd.h
+++ b/include/linux/genhd.h
@@ -85,6 +85,7 @@ struct disk_stats {
 	unsigned long ticks[2];
 	unsigned long io_ticks;
 	unsigned long time_in_queue;
+	unsigned long data_sectors;
 };
 
 #define PARTITION_META_INFO_VOLNAMELTH	64
-- 
2.11.0

